### PR TITLE
[#269] Repeat question groups

### PR DESCRIPTION
**Pull request by muloem**
* Set a question group as repeatable
* Render repeat question groups on the dashboard
* Generate reports with data for repeat question groups

### CODE REVIEW

File: Dashboard/app/js/lib/controllers/data-controllers.js [hunk 1]
```diff
@@ -312,11 +312,78 @@ FLOW.surveyedLocaleControl = Ember.ArrayController.create({
...
+            maxIter = item.get('iteration');
+          }
+        });
+
+        for (var iter = 0; iter <= maxIter; iter++){
+          var iterationAnswers = answersInGroup.filter(function(answer) {
+            return answer.get('iteration') === iter;
+          });
+
+          // TODO: +2 is a workaround for the fact that iterations currently start at -1
```
Comment (ichinaski) [inline Dashboard/app/js/lib/controllers/data-controllers.js]: Is this referring to the bug we hunted yesterday? If so, all iteration IDs are guaranteed to be 0+.

File: Dashboard/app/js/lib/controllers/data-controllers.js [hunk 2]
```diff
@@ -312,11 +312,78 @@ FLOW.surveyedLocaleControl = Ember.ArrayController.create({
...
+          iterationAnswers.iteration = iter+2;
+
+          groupResponses.push(iterationAnswers);
+
```
Comment (ichinaski) [inline Dashboard/app/js/lib/controllers/data-controllers.js]: [Fussy formatting comment] Extra line.

File: Dashboard/app/js/lib/views/data/question-answer-view.js [hunk 3]
```diff
@@ -1,138 +1,141 @@
...
     }
-    if ((this.get('isPhotoType') || this.get('isVideoType')) && !Ember.empty(this.content.get('value'))) {
-      // Since photos have a leading path from devices that we need to trim
-      this.set('photoUrl', FLOW.Env.photo_url_root + this.content.get('value').split('/').pop());
+  }.property('this.question'),
+
+  question: function(){
+    var c = this.get('content');
+    if (c) {
+      var questionId = this.content.get('questionID');
```
Comment (ichinaski) [inline Dashboard/app/js/lib/views/data/question-answer-view.js]: Given my little knowledge on Ember, this is more of a naive question, but isn't `this.content` pointing to the same variable as the previously declared `c`?

Comment (muloem) [inline Dashboard/app/js/lib/views/data/question-answer-view.js]: Yep, completely right. Will sort this one out as well.

**Comment by ichinaski** [comment]
@muloem latest changes look good! :+1:

**Comment by muloem** [comment]
:+1: code reviewed

**Pull request merged**